pkg/console/console_unix.go

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+//go:build !windows
+// +build !windows
+
+package console
+
+import (
+	"os"
+	"golang.org/x/term"
+)
+
+// isStdinRedirected checks if stdin is coming from a pipe or redirection
+func isStdinRedirected() bool {
+	stat, err := os.Stdin.Stat()
+	if err != nil {
+		// If we can't determine, assume it's not redirected
+		return false
+	}
+	
+	// If it's not a character device, it's coming from a pipe or redirection
+	if (stat.Mode() & os.ModeCharDevice) == 0 {
+		return true
+	}
+	
+	// Double-check using isatty
```

Comment:
the comment says `isatty` but the code uses IsTerminal. This file is identical to console_windows.go. If the code for both platforms is identical, there's no need for separate code files.